partiql-ast/src/main/resources/partiql_ast.ion

```diff
@@ -785,23 +788,27 @@ returning::{
 // `( CONSTRAINT <column_constraint_name> )?  <column_constraint_def>`
 table_definition::{
   columns: list::[column],
+  // table level constraints
+  constraints: list::[constraint],
   _: [
     column::{
-      name:        string,
+      name:        '.identifier.symbol',
       type:        '.type',
       constraints: list::[constraint],
-      _: [
-        // TODO improve modeling language to avoid these wrapped unions
-        // Also, prefer not to nest more than twice
-        constraint::{
-          name: optional::string,
-          body: [
-            nullable::{},
-            not_null::{},
-            check::{ expr: expr },
-          ],
-        },
-      ],
+    }
+  ],
+}
+
+constraint::{
+  name: optional::string,
```

Comment:
This is a good call out. 

Sql's grammar is constructed this way because they treat constraint as an object... 

The semantics with constraint name is not clear at the moment, but constraint name was modeled as a string previously in our grammar.